### ISSUE TITLE
aspire or vcarve pp not processing the tool paths properly 

### ISSUE THREAD

**Issue by gysnst2** [Snapmaker/snapmaker_cnc_post_process#19]
I doing some vcarve practice files.  and when I set the work origin, the ran my boundary every this was great. I ran the file and it started to cut circles.  not the file. I let is run for a bit until it started to cut threw my snapmaker hold-downs. i stopped it and loaded the gcode into luban and omg I was surprised at what I saw.  so then I reinstalled the postprocessor and recalculated the file and loaded it into luban and now at least it shows it being a good process it is not cutting the entire file.  I have tried to recalculate the file several time with different outcomes. 

![image](https://user-images.githubusercontent.com/32149597/115972082-06fa6680-a512-11eb-9931-99eb4a2e07c3.jpeg)

![image](https://user-images.githubusercontent.com/32149597/115972097-124d9200-a512-11eb-8621-df9ff3b4ba24.jpeg)

![image](https://user-images.githubusercontent.com/32149597/115972103-17124600-a512-11eb-81c3-26eb88398477.jpeg)

![image](https://user-images.githubusercontent.com/32149597/115972106-1e395400-a512-11eb-9d7c-4085bfabe7aa.jpeg)

**Comment by gysnst2** [comment]
I swear no-one reads these

**Comment by MickeEKlund** [comment]
**So, I got this solved! :-)**   

I'm using VCarve Desktop 10.514. The curves looks like a mess in Luban but cuts correctly, just transfer to SnapMaker and go (do yout boundry check first, just in case).

I took the PostProcessor for Aspire (not vCarve) and changed two lines:

VAR ARC_CENTRE_I_ABS_POSITION = [I|A| I|1.3]
VAR ARC_CENTRE_J_ABS_POSITION = [J|A| J|1.3]

I replaced INC with ABS and so far it has turned out great! Works like a charm! A small sign that took 4 hours yesterday has run in like 30 minutes!

PP-File: [https://forum.snapmaker.com/uploads/short-url/eaPc8GWblYMozc5sPftLI8Lnk18.zip](url)
Thread on Snapmaker Forum: [https://forum.snapmaker.com/t/cnc-vcarve-desktop-start-and-stop/15908](url)

**Comment by gysnst2** [comment]
Boundary checks are always good but then it cut threw my snapmaker hold down.  And in vetric it shows a full shogun being cut but when it loaded in LuBan only part of the project is being carved.  Or it's crazy carving circles

**Comment by MickeEKlund** [comment]
My did too, but after replacing the PP file it’s working 100%

BR
Mike

________________________________

Mikael Eklund
Kesuddens Konsult AB
Röksvampsgatan 17
749 42 Enköping
***@***.******@***.***>
0702-600 860

Från: gysnst2 ***@***.***>
Skickat: den 28 april 2021 15:01
Till: Snapmaker/snapmaker_cnc_post_process ***@***.***>
Kopia: Mikael Eklund ***@***.***>; Comment ***@***.***>
Ämne: Re: [Snapmaker/snapmaker_cnc_post_process] aspire or vcarve pp not processing the tool paths properly (#19)


Boundary checks are always good but then it cut threw my snapmaker hold down. And in vetric it shows a full shogun being cut but when it loaded in LuBan only part of the project is being carved. Or it's crazy carving circles

**Comment by artemician** [comment]
Sorry I didn't see this soon enough or I could've helped. I use aspire no problem. When I got the post processor file off GitHub through the GitHub desktop, for whatever reason moving the file from the default GitHub folder into the aspire folder corrupts it and causes the pp file to not work right.

As far as vCarve goes, vCarve is a segment taken from aspire so that if people only need the vCarve features and not the entire Aspire collection or want to buy the full version of vCarve without the high price tag of Aspire, then they can. There should be no difference between doing this project through vCarve vs Aspire unless some settings you didn't know about weren't set correctly because, again, Aspire includes everything whereas vCarve is just a segment taken from Aspire.

I bought Aspire and have spent months learning, tweaking, tuning and practicing to make it run perfectly with the Snapmaker 2.0 and once the learning process is done, Aspire works really well for the 2.0.

Aspire can be overwhelming at first because of the sheer amount of features and what it can do. If you're new to CNC it's a good idea to start with the different standalone programs (vCarve, Cut3d, Cut2d etc) so you don't get confused as to what feature is best for what project, then ease into Aspire if you want to do more.

**Comment by gysnst2** [comment]
> **So, I got this solved! :-)**   
> 
> 
> 
> I'm using VCarve Desktop 10.514. The curves looks like a mess in Luban but cuts correctly, just transfer to SnapMaker and go (do yout boundry check first, just in case).
> 
> 
> 
> I took the PostProcessor for Aspire (not vCarve) and changed two lines:
> 
> 
> 
> VAR ARC_CENTRE_I_ABS_POSITION = [I|A| I|1.3]
> 
> VAR ARC_CENTRE_J_ABS_POSITION = [J|A| J|1.3]
> 
> 
> 
> I replaced INC with ABS and so far it has turned out great! Works like a charm! A small sign that took 4 hours yesterday has run in like 30 minutes!
> 
> 
> 
> PP-File: [https://forum.snapmaker.com/uploads/short-url/eaPc8GWblYMozc5sPftLI8Lnk18.zip](url)
> 
> Thread on Snapmaker Forum: [https://forum.snapmaker.com/t/cnc-vcarve-desktop-start-and-stop/15908](url)
> 
> 

Do you have a copy of your new set of files.  Also when I click on your links I get 404 error

**Comment by gysnst2** [comment]
> Sorry I didn't see this soon enough or I could've helped. I use aspire no problem. When I got the post processor file off GitHub through the GitHub desktop, for whatever reason moving the file from the default GitHub folder into the aspire folder corrupts it and causes the pp file to not work right.
> 
> As far as vCarve goes, vCarve is a segment taken from aspire so that if people only need the vCarve features and not the entire Aspire collection or want to buy the full version of vCarve without the high price tag of Aspire, then they can. There should be no difference between doing this project through vCarve vs Aspire unless some settings you didn't know about weren't set correctly because, again, Aspire includes everything whereas vCarve is just a segment taken from Aspire.
> 
> I bought Aspire and have spent months learning, tweaking, tuning and practicing to make it run perfectly with the Snapmaker 2.0 and once the learning process is done, Aspire works really well for the 2.0.
> 
> Aspire can be overwhelming at first because of the sheer amount of features and what it can do. If you're new to CNC it's a good idea to start with the different standalone programs (vCarve, Cut3d, Cut2d etc) so you don't get confused as to what feature is best for what project, then ease into Aspire if you want to do more.
> 
> 

I agree with you I have vcarve full version I paid a prety penny for it if I remember 1000$ or so.  I'm not new to vcarve. But I'm am struggling with LuBan and the vcarve postprocessor.  LuBan is not the worst but doesn't help us for cnc up until 3.14 LuBan just crashed for me.   I have used fusion with some success but fusion is not the workflow I like vcarve is pretty easy for me.  I use vcarve on my v1

**Comment by MickeEKlund** [comment]
Copy (the actual text, **not** right click and copy link) and paste the link in a new window and it will work...  [https://forum.snapmaker.com/uploads/short-url/eaPc8GWblYMozc5sPftLI8Lnk18.zip](url) 
Not sure why github breaks the link, I'm pasting it correctly I think....

**Comment by MickeEKlund** [comment]
[Snapmaker350ABS.zip](https://github.com/Snapmaker/snapmaker_cnc_post_process/files/6398291/Snapmaker350ABS.zip)

**Comment by gysnst2** [comment]
I used your zip file it made a tiny bit of difference 

Maybe pm me your messenger or something we can compare notes a little more

![image](https://user-images.githubusercontent.com/32149597/116616411-5077f680-a902-11eb-892d-c588681a8344.jpeg)
This is a 1/2 vbit pretty stock settings

![image](https://user-images.githubusercontent.com/32149597/116616494-64235d00-a902-11eb-844a-d59771387d2e.jpeg) this one is with a 1/2 vbit .25" dept set their is no reason for it not to cave the entire sign

**Comment by MickeEKlund** [comment]
My Luban look like a mess too but the SnapMaker cuts perfect anyways...  If I open the file in NCviewer it look OK!

I just load the G-code into Luban and sends it via WiFi anyways. Cuts just like it looks in NCViewer!

![2021-04-30 13_08_27-2 - Finlir v bit cnc - NCViewer](https://user-images.githubusercontent.com/83349635/116687441-7ad6bc00-a9b5-11eb-932b-87149b3670e6.png)
![2021-04-30 13_10_09-Snapmaker Luban 3 14 0 _ new](https://user-images.githubusercontent.com/83349635/116687447-7dd1ac80-a9b5-11eb-9476-57a44173ed11.png)

**Comment by gysnst2** [comment]
![image](https://user-images.githubusercontent.com/32149597/116946585-1ed39800-ac40-11eb-8fc8-1c3a864399bf.jpeg)

![image](https://user-images.githubusercontent.com/32149597/116946597-2dba4a80-ac40-11eb-9ebb-caf80fba3911.jpeg)

![image](https://user-images.githubusercontent.com/32149597/116946618-3c086680-ac40-11eb-8d81-85af86792a37.jpeg)


I tried the profile it's a bit better but doesn't fix my issues.

**Comment by pistrak** [comment]
Я установил в Aspire пост процесс и и мой snapmaker a350 заработал.
![photo_2024-09-13_22-56-14](https://github.com/user-attachments/assets/d8bbef8e-152f-42bc-8c07-65f1eeeb838b)

**Comment by pistrak** [comment]
[https://forum.snapmaker.com/uploads/short-url/eaPc8GWblYMozc5sPftLI8Lnk18.zip](https://github.com/Snapmaker/snapmaker_cnc_post_process/issues/url)
